vyper/context/types/function.py

```diff
@@ -350,6 +359,9 @@ def fetch_call_return(self, node: vy_ast.Call) -> Optional[BaseTypeDefinition]:
             kwarg_keys += ["gas", "value"]
         validate_call_args(node, self.arg_count, kwarg_keys)
 
+        if "value" in [k.arg for k in node.keywords] and self.mutability < StateMutability.PAYABLE:
+            raise CallViolation("Cannnot call nonpayable functions with 'value' set", node)
```

Comment:
```suggestion
        if self.mutability < StateMutability.PAYABLE:
            kwarg_node = next((k for k in node.keywords if k.arg == "value"), None)
            if value_node is not None:
                raise CallViolation("Cannnot send ether to nonpayable function", kwarg_node)
```

This makes the exception annotation point directly at `value=` so the mistake is a bit clearer.